src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -0,0 +1,222 @@
+import com.powsybl.commons.reporter.Report;
+import com.powsybl.commons.reporter.ReportBuilder;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.TypedValue;
+import com.powsybl.iidm.modification.scalable.Scalable;
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Component;
+import com.powsybl.iidm.network.DefaultNetworkListener;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.IdentifiableType;
+import com.powsybl.iidm.network.Load;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.extensions.GeneratorStartup;
+import org.gridsuite.modification.server.NetworkModificationException;
+import org.gridsuite.modification.server.dto.GenerationDispatchInfos;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import static java.util.Comparator.comparingInt;
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toCollection;
+import static org.gridsuite.modification.server.NetworkModificationException.Type.GENERATION_DISPATCH_ERROR;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class GenerationDispatch extends AbstractModification {
+    private static final String COMPONENT = "CC";
+    private static final String POWER_TO_DISPATCH = "PowerToDispatch";
+    private static final String STACKING = "Stacking";
+    private static final String RESULT = "Result";
+
+    private final Map<Integer, List<Generator>> fixedSupplyGenerators = new HashMap<>();
+    private final Map<Integer, List<Generator>> adjustableGenerators = new HashMap<>();
+
+    private final Map<Integer, Double> totalDemand = new HashMap<>();
+    private final Map<Integer, Double> remainingPowerImbalance = new HashMap<>();
+
+    private final GenerationDispatchInfos generationDispatchInfos;
+
+    public GenerationDispatch(GenerationDispatchInfos generationDispatchInfos) {
+        this.generationDispatchInfos = generationDispatchInfos;
+    }
+
+    private void report(Reporter reporter, String key, String defaultMessage, Map<String, Object> values, TypedValue severity) {
+        ReportBuilder builder = Report.builder()
+            .withKey(key)
+            .withDefaultMessage(defaultMessage)
+            .withSeverity(severity);
+        for (Map.Entry<String, Object> valueEntry : values.entrySet()) {
+            builder.withValue(valueEntry.getKey(), valueEntry.getValue().toString());
+        }
+        reporter.report(builder.build());
+    }
+
+    private double computeTotalActiveLoad(Component component) {
+        Objects.requireNonNull(component);
+        final AtomicDouble totalLoad = new AtomicDouble(0.);
```

Comment:
AtomicDouble is not advisable here. I would go with
```
        OptionalDouble mayTotal = component.getBusStream().flatMap(Bus::getLoadStream)
            .filter(load -> load.getTerminal().isConnected())
            .mapToDouble(Load::getP0)
            .reduce(Double::sum);
        return mayTotal.orElse(0);
```

Reply:
Done